packages/server/src/fhir/operations/expand.test.ts

```diff
@@ -1288,24 +1288,66 @@ describe('Expand', () => {
     ]);
   });
 
-  test('Base resources are not shadowed for Super Admin', async () => {
-    const url = 'https://medplum.com/fhir/ValueSet/resource-types';
+  test('Honors ValueSet designation overrides', async () => {
+    const codeSystem: CodeSystem = {
+      resourceType: 'CodeSystem',
+      url: 'http://example.com/CodeSystem/' + randomUUID(),
+      content: 'example',
+      status: 'draft',
+      concept: [
+        {
+          code: 'MSG_INVALID_ID',
+          display: 'ID not accepted',
+          designation: [
+            { language: 'fr', value: 'ID non accepté' },
+            { language: 'zh', value: 'ID不被接受' },
+          ],
+        },
+      ],
+    };
+    const valueSet: ValueSet = {
+      resourceType: 'ValueSet',
+      status: 'draft',
+      url: 'https://example.com/ValueSet/' + randomUUID(),
+      compose: {
+        include: [
+          {
+            system: codeSystem.url,
+            concept: [
+              {
+                code: 'MSG_INVALID_ID',
+                display: 'Invalid ID',
+                designation: [
+                  { language: 'fr', value: 'Identifiant invalide' },
+                  { language: 'es', value: 'ID inválido' },
+                ],
+              },
+            ],
+          },
+        ],
+      },
+    };
     const csRes = await request(app)
       .post('/fhir/R4/CodeSystem')
       .set('Authorization', 'Bearer ' + accessToken)
-      .send({
-        resourceType: 'CodeSystem',
-        status: 'active',
-        url,
-        content: 'not-present',
-      } satisfies CodeSystem);
+      .send(codeSystem);
     expect(csRes.status).toStrictEqual(201);
+    const vsRes = await request(app)
+      .post('/fhir/R4/ValueSet')
+      .set('Authorization', 'Bearer ' + accessToken)
+      .send(valueSet);
+    expect(vsRes.status).toStrictEqual(201);
 
-    const superAdminToken = await initTestAuth({ superAdmin: true });
     const res = await request(app)
-      .get(`/fhir/R4/ValueSet/$expand?url=${url}&filter=clien`)
-      .set('Authorization', 'Bearer ' + superAdminToken);
-    expect(res.status).toBe(200);
-    expect(res.body.expansion.contains[0].display).toStrictEqual('ClientApplication');
+      .get(
+        `/fhir/R4/ValueSet/$expand?url=${encodeURIComponent(valueSet.url as string)}&filter=invalid&displayLanguage=fr`
```

Comment:
Non blocking: you have this type cast because ValueSet.url is typed as `string | undefined`, but since you are defining an explicit object above with this property set you can do a little better here! Rewriting the definition from a type declaration to a "satisfies" constraint will retain the narrower type and let you use it without a cast, while still validating that the object is assignable to the ValueSet type.

```
const valueSet = {
  url: 'https://example.com/ValueSet/' + randomUUID(),
  ...
} satisfies ValueSet;

encodeURIComponent(valueSet.url)

```